lib/check-args.js

```diff
@@ -1,6 +1,12 @@
 const debug = require('debug')('selenium-standalone:install');
 
-const checkArgs = (fnName, opts = {}) => {
+/**
+ * @template T
+ * @param {string} fnName
+ * @param {T} [opts]
+ * @returns {T}
```

Comment:
Can we update this or remove?

Reply:
Of course! What direction would you like to take?

I'll remove it for now. I found this function is the hardest part. Here's how it is used in start and install:

https://github.com/webdriverio/selenium-standalone/blob/57e1eb140f7029aa81d46452b87326f39e7c50ac/lib/install.js#L34-L37

Given an optional `_opts`, when we call `checkArgs('...', _opts)` we want to return that type (or Partial). I got stuck here with the `Partial<T>`:

```js
/**
 * @template T
 * @param {string} fnName
 * @param {T} opts
 * @returns {Partial<T>}
 */
const checkArgs = (fnName, opts = {}) => {
//                         ^^^^^^^^^
// Type '{}' is not assignable to type 'T'.
//  'T' could be instantiated with an arbitrary type which could be unrelated to '{}'.ts(2322)
```

Reply:
oh .. not sure how to do that with JSDoc, in TS this would look like:

```ts
const checkArgs<T extends object> = (fnName: string, opts: T = {}): T => {
  debug(fnName + ' called with', opts);

  return { ...opts };
};
```

Reply:
Yea, I wanted to work in JS so I could work in smaller steps. Would you be OK with the change knowing that it only changes what is logged in the `debug` call from `{}` to `undefined`? I added tests before I gave up on JSDoc. :laughing: 

--

I following https://github.com/microsoft/TypeScript/issues/41768 which says that:

```
<T extends object>
```
would become
```
 * @template {object} T
```

but TS infers it as `unknown`:

```
/**
 * @template {object} T
 * @param {string} fnName
 * @param {T} [opts]
 * @returns {T}
 */
const checkArgs = (fnName, opts = {}) => {
//                         ^^^^^^^^^
// Type '{}' is not assignable to type 'T'.
//  '{}' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'unknown'.ts(2322)
```

It is similar with `@template {object} T` but you get `{}` instead of `unknown`:

```
Type '{}' is not assignable to type 'T'.
  '{}' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint '{}'.ts(2322)
```

That's where I gave up, added a test, and removed ` = {}`. We could also inline the method to remove the problem.

Reply:
Yeah, this should not blocking us from moving forward.